File: coding-patterns/01-two-pointers/010.py
# Pair with Target Sum (Easy)

class Solution:
  def search(self, arr, target_sum):
    left, right = 0, len(arr) - 1
    while(left < right):
      current_sum = arr[left] + arr[right]
      if current_sum == target_sum:
        return [left, right]

      if target_sum > current_sum:
        left += 1  
      else:
        right -= 1 
    return [-1, -1]

File: coding-patterns/01-two-pointers/010.cpp
// Pair with Target Sum (Easy)

using namespace std;

#include <iostream>
#include <vector>

class Solution {

  public:

  static vector<int> search(const vector<int> &arr, int targetSum) {
    // TODO: Write your code here
    int start = 0;
    for (int end = arr.size() - 1; end > -1; end--) {
      while (start < end && arr[start] + arr[end] < targetSum)
        start++;
      if (start != end && arr[start] + arr[end] == targetSum) {
        return vector<int>{start, end};
      }
    }
    return vector<int>(2, -1);
  }

};
